### ISSUE TITLE
PyTorchYolo estimator affects model parameters during loss/gradient calculation

### ISSUE THREAD

**Issue by beat-buesser** [Trusted-AI/adversarial-robustness-toolbox#1859]
**Describe the bug**
To obtain the loss the Yolo model the estimator PyTorchYolo has to set to model to mode train(). But this also lets the batch-norm layer of the model use batch statistics and update the tracked statistics which changes the model parameters and prediction accuracy of the model. Solution is to set the batch-norm layers back to eval model after the call to train().

**To Reproduce**
Run repeatedly `compute_loss` directly or as part of an evasion attack.
